internal/provider/provider.go

```diff
@@ -58,14 +58,14 @@ func New(version string) func() *schema.Provider {
 					Type:        schema.TypeString,
 					Optional:    true,
 					DefaultFunc: schema.EnvDefaultFunc("CLOUDFLARE_EMAIL", nil),
-					Description: "A registered Cloudflare email address. Alternatively, can be configured using the `CLOUDFLARE_EMAIL` environment variable.",
+					Description: "A registered Cloudflare email address. Alternatively, can be configured using the `CLOUDFLARE_EMAIL` environment variable. An email address for API authentication is only necessary when used with an API Key which is now considered as legacy by Cloudflare, it is not required with API tokens anymore.",
```

Comment:
let's use [`ConflictsWith`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#section-readme) instead. this will automatically append a warning from https://github.com/cloudflare/terraform-provider-cloudflare/pull/1727/files#diff-58d6a027753b50994deb7e11e4a99dde423f35844986019bd9cea5e0c94aba22L46.

```suggestion
					Description: "A registered Cloudflare email address. Alternatively, can be configured using the `CLOUDFLARE_EMAIL` environment variable.",
					ConflictsWith: []string{"api_token"},
```